### ISSUE TITLE
getwininfo(winid) returns wrong winnr

### ISSUE THREAD

**Issue by liushapku** [neovim/neovim#9277]
- `nvim --version`: 0.3.1
- Vim (version: ) behaves differently? don't know
- Operating system/version: ubuntu
- Terminal name/version:
- `$TERM`:

### Steps to reproduce using `nvim -u NONE`

```
nvim -u NONE

:split
:split
:echo getwininfo(1000)[0].winnr
:echo getwininfo(1001)[0].winnr
:echo getwininfo(1002)[0].winnr
```

### Actual behaviour
shows 1 three times

### Expected behaviour
shows 1 2 3

**Comment by liushapku** [comment]
a pull request has been created [#9278](https://github.com/neovim/neovim/pull/9278)

**Comment by janlazo** [comment]
vim-patch:8.1.0187 wasn't merged.

**Comment by jamessan** [comment]
> vim-patch:8.1.0187 wasn't merged.

That's not the problem.  I mis-merged v7.4.2215.
